src/main/java/imsam/mgen/MultiTargetModelGenerator.java

```diff
@@ -0,0 +1,355 @@
+                            if(targetCandidate>numberOfStates-1 || targetCandidate==0){
+                                String errMsg = "Targets must be in the state space and cannot be the initial state";
+                                logger.error(errMsg);
+                                System.exit(1);
+                            } else {
+                                logger.debug("-------------adding "+ targetCandidate+ " to target ArrayList-------------");
+                                targets.add(targetCandidate);
+                            }
+                        }catch(NumberFormatException e){
+                            String errMsg = "The target state list must be made up of integers separated by a space. "+
+                            "Ex)\"1 5 9 12\"";
+                            logger.error(errMsg);
+                            System.exit(1);
+                        }
+                    }
+                    targets.add(targetState);
+                }
+            }
+        } else {
+            //if absorb is selected then the target list will only contain the target state
+            targets = new ArrayList<>();
+            targets.add(targetState);
+        }
+    }
+
+   
+
+    /**
+     * Generates a model by creating paths from target states to the initial state
+     * It then ensures every state connects to a target state path
+     * Finally it randomly adds transitions
+     */
+    @Override
+    protected void generateModel(){
+        long start = System.nanoTime(); //to calculate runtime
+
+        //initializing state space
+        stateSpace = new State[numberOfStates];
+        logger.debug("-------------Initializing state space-------------");
+        for(int stateId=0; stateId < numberOfStates; stateId++){
+            stateSpace[stateId] = new State(stateId);
+        }
+
+        /*
+         * This sections creates paths from the initial state to target states.
+         * The method is to start at the target states and randomly generate predecessor states.
+         * A check is done to ensure that the predecessor is not already on the path and is not a target state.
+         * Predecessors are tracked using the TargetPath class.
+        */
+        logger.debug("-------------Generating target path(s)-------------");
+        targetPathTracker = new TargetPath[targets.size()];
+        boolean targetPathCheck[] = new boolean[numberOfStates];
+        int initial = 0;
+        for(int targetIdx = 0; targetIdx < targets.size(); targetIdx++){
+            int currentTarget = targets.get(targetIdx);
+            logger.trace("Starting path for "+ currentTarget);
+            targetPathTracker[targetIdx] = new TargetPath(currentTarget);
+            logger.trace("Setting targetPathCheck["+currentTarget+"] to true");
+            targetPathCheck[currentTarget] = true;
+            int current = currentTarget;
+            int count = 1;
+            while(current != initial){
```

Comment:
I feel like it might be more readable if it's a for loop that looks like
`for(int count = 1; current != initial; count++)`
But that's just my preference.

Reply:
Thanks, change made